core/src/main/java/com/novoda/merlin/MerlinsBeard.java

```diff
@@ -44,16 +48,34 @@ private NetworkInfo getNetworkInfo() {
     }
 
     /**
-     * Provides a boolean representing whether a wifi network connection has been established.
+     * Provides a boolean representing whether a Wi-Fi network connection has been established.
      *
      * NOTE: Therefore available does not necessarily mean that an internet connection
      * is available.
      *
-     * @return boolean true if a wifi network connection is available.
+     * @return boolean true if a Wi-Fi network connection is available.
      */
     public boolean isConnectedToWifi() {
         NetworkInfo networkInfo = connectivityManager.getNetworkInfo(ConnectivityManager.TYPE_WIFI);
         return networkInfo != null && networkInfo.isConnected();
     }
 
+
+    /**
+     * Provides a boolean representing whether a mobile network connection has been established.
+     * NOTE: Therefore available does not necessarily mean that an internet connection
+     * is available.
+     *
+     * @return boolean true if a mobile network connection is available.
+     */
+    public boolean isConnectedToMobileNetwork() {
+        int simState = telephonyManager.getSimState();
+        if (simState == TelephonyManager.SIM_STATE_READY) {
```

Comment:
What you describe is surely possible, but only in case of network type we are not considering (like *bluetooth* or *ethernet*) being connected. Otherwise if the sim is not ready you cannot be connected to the mobile network